### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by q7ice**

### CODE REVIEW

**Comment by andrewzenkov** [comment]
It needs to be fixed since we have not soft ui if there a lot of tasks

File: src/components/Filter/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
```
Comment (andrewzenkov) [inline src/components/Filter/Filter.jsx:1]: Forgot to make export in Filter/index.js (now it's empty file )

File: src/components/App/todoSlice.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      obj.text = action.payload;
+      obj.done = false;
+      state.todos.push(obj);
+    },
+    deleteTodo: (state, action) => {
+      state.todos = state.todos
+        .filter((todo) => todo.id !== action.payload)
+        .map((todo, index) => {
+          todo.id = index;
+          return todo;
```
Comment (andrewzenkov) [inline src/components/App/todoSlice.js:34]: I would recommend don't handle Id of todos as index or something like that.
Please, use library to generate random id and set if for todos once.

It's always bad to change ids during application working

File: src/components/InputTodo/InputTodo.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        type="text"
+        placeholder="New todo..."
+        autoFocus
+        value={todoInput}
+        onChange={(e) => setTodoInput(e.target.value)}
+        onKeyDown={(e) => {
+          if (e.key === "Enter") {
+            addingTodo();
+          }
+        }}
```
Comment (andrewzenkov) [inline src/components/InputTodo/InputTodo.jsx:30]: In variables as I mentioned

File: src/components/App/todoSlice.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+};
+export const todoSlice = createSlice({
+  name: "todo",
+  initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const obj = {};
+      obj.id = state.todos.length;
+      obj.text = action.payload;
+      obj.done = false;
```
Comment (andrewzenkov) [inline src/components/App/todoSlice.js:26]: I would recommend to use Model for any entity
at least you can set up exact field you need to fill and set up as fully separated entity but not an usual object

```
      const todo = new Todo({
        id: state.todos.length,
        text: action.payload,
        done: false
      });

      state.todos.push(obj);
```

**Review by andrewzenkov — CHANGES_REQUESTED** [review]

File: src/components/App/todoSlice.js (hunk 2 above)
Comment (q7ice) [reply src/components/App/todoSlice.js:34]: Which libraries would you recommend for such things?

File: src/components/App/todoSlice.js (hunk 4 above)
Comment (q7ice) [reply src/components/App/todoSlice.js:26]: Yeah, I found out about this after reading the style guide from "Airbnb", but didn't change the project

File: src/components/TodoItem/TodoItem.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import CheckMarkIcon from "../Icons/CheckMarkIcon";
+import "./TodoItem.scss";
+
+function TodoItem({ text, todoId, done }) {
+  const dispatch = useDispatch();
+  return (
+    <div className="todo-item">
+      <div className={done ? "todo-done" : null}>{text}</div>
+      <button 
+        onClick={() => dispatch(toggleDone(todoId))}
```
[the remark this answers is not in the document]
Comment (q7ice) [reply src/components/TodoItem/TodoItem.jsx:15]: Did you mean that I need to move this code "() => dispatch(toggleDone(todoId))" to variable or I didn't understand you?

File: src/components/TodoItem/TodoItem.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import DeleteIcon from "../Icons/DeleteIcon";
+import CrossIcon from "../Icons/CrossIcon";
+import CheckMarkIcon from "../Icons/CheckMarkIcon";
+import "./TodoItem.scss";
+
+function TodoItem({ text, todoId, done }) {
+  const dispatch = useDispatch();
+  return (
+    <div className="todo-item">
+      <div className={done ? "todo-done" : null}>{text}</div>
```
[the remark this answers is not in the document]
Comment (q7ice) [reply src/components/TodoItem/TodoItem.jsx:13]: Do I need to move this {done ? "todo-done" : null} to new variable?

File: src/components/App/todoSlice.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import { createSlice } from "@reduxjs/toolkit";
```
[the remark this answers is not in the document]
Comment (q7ice) [reply src/components/App/todoSlice.js:1]: I will use that next time, just didn't know it then :)

File: src/components/Filter/Filter.jsx (hunk 1 above)
Comment (q7ice) [reply src/components/Filter/Filter.jsx:1]: Oops

**Comment by q7ice** [comment]
> 
> 
> It needs to be fixed since we have not soft ui if there are a lot of tasks

I noticed that problem while was developing App, but I forgot to fix that after work. I can put all the "TODOS" in a smaller container, which will have a height lower than the input Todo to fix this problem.

File: src/components/TodoItem/TodoItem.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (andrewzenkov) [reply src/components/TodoItem/TodoItem.jsx:15]: Exactly

File: src/components/TodoItem/TodoItem.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (andrewzenkov) [reply src/components/TodoItem/TodoItem.jsx:13]: No, please check 'classnames' library

File: src/components/App/todoSlice.js (hunk 2 above)
Comment (andrewzenkov) [reply src/components/App/todoSlice.js:34]: 1) you can use reselect
2) but at least you can just move it in selector

**Review by andrewzenkov — APPROVED** [review]

**Pull request merged**